WoWsShipBuilder.ViewModels/Other/ShipComparisonViewModel.cs

```diff
@@ -0,0 +1,762 @@
+    {
+        UseUpgradedModules = !UseUpgradedModules;
+        await ChangeModulesBatch();
+    }
+
+    public async Task ToggleHideShipsWithoutSelectedSection()
+    {
+        HideShipsWithoutSelectedSection = !HideShipsWithoutSelectedSection;
+        await ApplyFilters();
+    }
+
+    private List<ShipComparisonDataWrapper> HideShipsIfNoSelectedSection(IEnumerable<ShipComparisonDataWrapper> list)
+    {
+        if (!HideShipsWithoutSelectedSection)
+        {
+            return list.ToList();
+        }
+
+        List<ShipComparisonDataWrapper> newList = SelectedDataSection switch
+        {
+            ShipComparisonDataSections.MainBattery => list.Where(x => x.ShipDataContainer.MainBatteryDataContainer is not null).ToList(),
+            ShipComparisonDataSections.He => list.Where(x => x.HeDamage is not null).ToList(),
+            ShipComparisonDataSections.Ap => list.Where(x => x.ApDamage is not null).ToList(),
+            ShipComparisonDataSections.Sap => list.Where(x => x.SapDamage is not null).ToList(),
+            ShipComparisonDataSections.Torpedo => list.Where(x => x.ShipDataContainer.TorpedoArmamentDataContainer is not null).ToList(),
+            ShipComparisonDataSections.RocketPlanes => list.Where(x => x.RocketPlanesType.Any()).ToList(),
+            ShipComparisonDataSections.Rockets => list.Where(x => x.RocketPlanesWeaponType.Any()).ToList(),
+            ShipComparisonDataSections.TorpedoBombers => list.Where(x => x.TorpedoBombersType.Any()).ToList(),
+            ShipComparisonDataSections.AerialTorpedoes => list.Where(x => x.TorpedoBombersWeaponType.Any()).ToList(),
+            ShipComparisonDataSections.Bombers => list.Where(x => x.BombersType.Any()).ToList(),
+            ShipComparisonDataSections.Bombs => list.Where(x => x.BombersWeaponType.Any()).ToList(),
+            ShipComparisonDataSections.Sonar => list.Where(x => x.ShipDataContainer.PingerGunDataContainer is not null).ToList(),
+            ShipComparisonDataSections.SecondaryBattery => list.Where(x => x.ShipDataContainer.SecondaryBatteryUiDataContainer.Secondaries is not null).ToList(),
+            ShipComparisonDataSections.SecondaryBatteryShells => list.Where(x => x.ShipDataContainer.SecondaryBatteryUiDataContainer.Secondaries is not null).ToList(),
+            ShipComparisonDataSections.AntiAir => list.Where(x => x.ShipDataContainer.AntiAirDataContainer is not null).ToList(),
+            ShipComparisonDataSections.AirStrike => list.Where(x => x.ShipDataContainer.AirstrikeDataContainer is not null).ToList(),
+            ShipComparisonDataSections.Asw => list.Where(x => x.ShipDataContainer.AswAirstrikeDataContainer is not null || x.ShipDataContainer.DepthChargeLauncherDataContainer is not null).ToList(),
+            _ => list.ToList(),
+        };
+
+        newList.AddRange(PinnedShipList.Where(x => !ContainsWrapper(x, newList)));
+
+        return newList;
+    }
+
+    public List<ShipComparisonDataSections> GetDataSectionsToDisplay()
```

Comment:
When exactly is this method called? Ensure its results are properly cached and it is invoked only when necessary